src/main/java/de/tum/in/www1/artemis/security/jwt/TokenProvider.java

```diff
@@ -151,25 +153,31 @@ public Authentication getAuthentication(String token) {
     }
 
     /**
-     * Checks if a certain downloadFileKey and filename is inside the JWT token. Also validates the JWT token if it is still valid.
+     * Checks if custom claims are inside the signed JWT token. Also validates the JWT token if it is still valid.
      *
      * @param authToken The token that has to be checked
-     * @param downloadFileKey What downloadFileKey should be included in the token
-     * @param fileName The name of the file the token belongs to
+     * @param requiredClaims The claims that should be included in the token
      * @return true if everything matches
      */
-    public boolean validateTokenForAuthorityAndFile(String authToken, String downloadFileKey, String fileName) {
-        if (!validateJwsToken(authToken)) {
+    public boolean validateTokenForAuthorityAndFile(String authToken, Claims requiredClaims) {
+        if (!validateJwsToken(authToken) || requiredClaims.isEmpty()) {
             return false;
         }
-        try {
-            String fileNameToken = (String) parseClaims(authToken).get(FILENAME_KEY);
-            return fileNameToken.contains(downloadFileKey + fileName);
-        }
-        catch (Exception e) {
-            log.warn("Invalid action validateTokenForAuthorityAndFile: ", e);
+
+        Claims tokenClaims = parseClaims(authToken);
+
+        for (var requiredClaim : requiredClaims.entrySet()) {
+            try {
+                if (!tokenClaims.get(requiredClaim.getKey()).equals(requiredClaim.getValue())) {
+                    return false;
+                }
+            }
+            catch (NullPointerException e) {
+                log.warn("Received a null value as key of requiredClaim in validateTokenForAuthorityAndFile: ", e);
+                return false;
+            }
         }
```

Comment:
We should not catch NullPointerExceptions, instead the code should be null-save.
```suggestion
        for (var requiredClaim : requiredClaims.entrySet()) {
            if (!Objects.equals(tokenClaims.get(requiredClaim.getKey()), requiredClaim.getValue())) {
                return false;
            }
        }
```

Reply:
Adapted